### PR TITLE
Simple tool to analyze memory usage of bloop running with certain JVM…

### PR DESCRIPTION
**Pull request by tpasternak**
… options

### CODE REVIEW

File: gcbenchmark/gcbenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  env: Map[String, String],
+  maxTime: Duration,
+  avarageTime: Duration,
+  maxMemoryFootprintMb: Int,
+  idleMemoryFootprintMb: Int
+)
+
+object Main {
+  val workspace =
+    os.temp.dir(os.pwd, "tmp-", deleteOnExit = true) // where the temporary files are stored
```
Comment (lwronski) [inline gcbenchmark/gcbenchmark.scala:20]: I guess this directory should be removed at the end

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,6 +13,29 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
 
 jobs:
+  bloop-memory-footprint:
```
Comment (lwronski) [inline .github/workflows/ci.yml:16]: For now, let's not run the benchmark every time, maybe only in master.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -13,6 +13,29 @@ concurrency:
...
+        fetch-depth: 0
+        submodules: true
+    - uses: coursier/cache-action@v6.3
+    - uses: coursier/setup-action@v1.1.2
+      with:
+        jvm: 11
+    - name: Build Launcherr
+      run: ./mill -i cli.launcher
+    - name: Build Benchmark
+      run: ./out/cli/launcher/dest/launcher package --assembly gcbenchmark/gcbenchmark.scala
```
Comment (lwronski) [inline .github/workflows/ci.yml]: To avoid hardcode path to launcher, we can use `copyJvmLauncher` from `build.sc`

Comment (tpasternak) [reply .github/workflows/ci.yml]: It copies standalone launcher, for which I'm getting:
```
Exception in thread "main" java.lang.NoSuchMethodError: 'java.io.File coursier.paths.CachePath.defaultArchiveCacheDirectory()'
	at coursier.cache.CacheDefaults$.archiveCacheLocation$lzycompute(CacheDefaults.scala:18)
	at coursier.cache.CacheDefaults$.archiveCacheLocation(CacheDefaults.scala:18)
	at coursier.cache.ArchiveCache$.apply(ArchiveCache.scala:141)
	at coursier.jvm.JvmCache$.apply(JvmCache.scala:26)
	at scala.build.options.BuildOptions.javaHomeManager$lzycompute(BuildOptions.scala:240)
	at scala.build.options.BuildOptions.javaHomeManager(BuildOptions.scala:237)
	at scala.cli.commands.SharedOptions.bloopRifleConfig(SharedOptions.scala:189)
	at scala.cli.commands.Package$.run(Package.scala:45)
	at scala.cli.commands.Package$.run(Package.scala:31)
	at caseapp.core.app.CaseApp.main(CaseApp.scala:149)
	at caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:116)
	at scala.cli.ScalaCli$.main0(ScalaCli.scala:173)
	at scala.cli.ScalaCli$.main(ScalaCli.scala:114)
	at scala.cli.ScalaCli.main(ScalaCli.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at coursier.bootstrap.launcher.Bootstrap.main(Bootstrap.java:109)
	at coursier.bootstrap.launcher.ResourcesLauncher.main(ResourcesLauncher.java:79)
```
So I'll keep the hardcoded output path for now

**Review by lwronski — APPROVED** [review]

**Pull request merged**